math/fibonacci_sum.cpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * @brief An algorithm to calculate the sum of Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$
+ * @details An algorithm to calculate the sum of Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$ where \f$\mathrm{F}(i)\f$
+ * denotes the i-th Fibonacci Number . Note that F(0) = 0 and F(1) = 1.
+ * The value of the sum is calculated using matrix exponentiation.
+ * @source https://stackoverflow.com/questions/4357223/finding-the-sum-of-fibonacci-numbers
+ * @author [Sarthak Sahu](https://github.com/SarthakSahu1009)
+ */
+
+#include <iostream> /// for std::cin and std::cout
+#include <cassert>  /// for assert
+#include <vector>   /// for std::vector
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+    /**
+     * @namespace fibonacci_sum
+     * @brief Functions for the sum of the Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$
+     */
+    namespace fibonacci_sum {
+
```

Comment:
Just a suggestion:
```suggestion
using matrix = std::vector<std::vector<int> >;
```
this is akin to the `typedef` in C. Now, throughout the code, you can use `math::fibonacci_sum::matrix` as the datatype.
Throughout the code, the dimensions of this matrix is 2x2. Hence, the `std::array` might be an efficient option to use.